components/raftstore/src/store/fsm/apply.rs

```diff
@@ -3836,6 +3836,17 @@ where
                         "target region is not found, drop proposals";
                         "region_id" => region_id
                     );
+                    // Invoking callback can release txn latch, if it's still leader, following
+                    // command may not read the writes of previous commands and break ACID. If
+                    // it's still leader, there are two possibility that mailbox is closed:
+                    // 1. The process is shutting down.
+                    // 2. The leader is destroyed. A leader won't propose to destroy itself, so
+                    //     it should either destroyed by older leaders or newer leaders. Leader
+                    //     won't respond to read until it has applied to current term, so no
+                    //     command will be proposed until command from older leaders have applied,
```

Comment:
Is there any command from txnkv will propose write directly without reading?

Reply:
/cc @sticnarf 

The statement is true for all versions. And for latest version, leader also won't propose any command before applying to current term to build correct pipeline.